altapay.php

```diff
@@ -124,27 +129,38 @@ function altapay_add_gateway( $methods ) {
 			}
 
 			// Check if file exists
-			$path    = $terminalDir . $terminal . '.class.php';
-			$tmpPath = $tmpDir . '/' . $terminal . '.class.php';
+			$terminal_class_file     = $terminalDir . $terminal . '.class.php';
```

Comment:
How is `$terminalDir` configured? And ware you sanitising `$terminal` ? An attacker my be able to access a system file or folder if not done properly.

Reply:
@emicha We are generating the class file using a template file `paymentClass.tpl `for each terminal that extends the WooCommerce base gateway class and saves it to the `terminals` directory within the plugin. If the terminals directory is not writable we save the class file to the system's tmp directory.

Also, we sanitize the terminal's data before saving it to the database, and `$terminal ` has the value fetched from the DB.

Reply:
@shahbaz-mehar How are you generated the value that goes in $terminalDir ?

Reply:
@emicha $terminalDir is the path of `terminals` directory inside our plugin. That includes each terminal's class file that are active.

Reply:
@emicha We are generating value (the class file) using a template file `paymentClass.tpl` for each terminal

Reply:
@emicha Also, there is no user involvement and the value can't be changed by any configurations, etc.